src/EditorFeatures/Core/Implementation/Suggestions/SuggestedAction.cs

```diff
@@ -149,30 +150,28 @@ public void Invoke(CancellationToken cancellationToken)
         private async Task InvokeWorkerAsync(
             Func<Document> getFromDocument, IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
-            this.AssertIsForeground();
+            AssertIsForeground();
             IEnumerable<CodeActionOperation> operations = null;
-
             // NOTE: As mentioned above, we want to avoid computing the operations on the UI thread.
             // However, for CodeActionWithOptions, GetOptions() might involve spinning up a dialog
             // to compute the options and must be done on the UI thread.
-            var actionWithOptions = this.CodeAction as CodeActionWithOptions;
-            if (actionWithOptions != null)
+            if (CodeAction is CodeActionWithOptions actionWithOptions)
             {
                 var options = actionWithOptions.GetOptions(cancellationToken);
                 if (options != null)
                 {
                     // ConfigureAwait(true) so we come back to the same thread as 
                     // we do all application on the UI thread.
                     operations = await GetOperationsAsync(actionWithOptions, options, cancellationToken).ConfigureAwait(true);
-                    this.AssertIsForeground();
```

Comment:
since we ConfigureAwait(true) here, how do we avoid deadlocking since you called 'Wait' above?